商品出品機能

shokoron opened the pull request:
#What
商品出品機能を作成

#Why
商品出品機能を実装するため

商品出品機能テスト画像
https://i.gyazo.com/72ce188bf8dec5ad20e3250f5d41a076.png

 ログイン状態の場合は、商品出品ページへ遷移できる動画
https://i.gyazo.com/5e5ceeaf66f44f4aefa7036c042c4269.gif

価格が入力されると同時に、販売手数料と販売利益が表示される動画
https://i.gyazo.com/1f0909575c55fda4bf50fc7e2cfc1aed.gif

必要な情報を適切に入力して「出品する」ボタンを押すと、商品情報がデータベースに保存される動画
https://i.gyazo.com/4db45f72c37d349e08b84f76feba97fe.gif

入力に問題がある状態で「出品する」ボタンが押された場合、情報は保存されず、出品ページに戻りエラーメッセージが表示される動画
https://i.gyazo.com/91777d300c9c6d54f0c03bd27b05b5ae.gif

 ログアウト状態の場合は、商品出品ページへ遷移しようとすると、ログインページへ遷移する動画
https://i.gyazo.com/cb898d2d4000881895494dfcbb20e4e7.gif

shokoron committed:
商品出品機能

shokoron committed:
モデル単体テスト

OhyaYouki reviewed (commented):
レビュー依頼ありがとうございます。

以下の動画にて確認できない点がございましたので再度撮影をお願いいたします。
```
ログアウト状態の場合は、商品出品ページへ遷移しようとすると、ログインページへ遷移する動画
https://i.gyazo.com/cb898d2d4000881895494dfcbb20e4e7.gif

「商品出品ページへ遷移しようとすると、ログインページへ遷移する」が確認できませんでした。
```

コードの修正がございましたらコミット・プッシュをお願いいたします。
修正後は再度レビュー依頼のご提出をお願いいたします。

shokoron committed:
修正しました

 ログアウト状態の場合は、商品出品ページへ遷移しようとすると、ログインページへ遷移する動画
https://i.gyazo.com/5f4478a2114cbc4a232b6d96e714c887.gif

OhyaYouki commented on app/controllers/items_controller.rb:
```diff
@@ -1,4 +1,29 @@
+  before_action :authenticate_user!, except: [:index, :show]
+
   def index
+    @items = Item.all
```
```ruby
@items = Item.all
```
こちらの記述は商品一覧機能の実装範囲となります。
商品一覧機能の実装時に差分として確認できなくなる可能性がございますので
現状ではコメントアウトもしくは削除しておきましょう！

OhyaYouki commented on spec/models/item_spec.rb:
```diff
@@ -0,0 +1,83 @@
+        expect(@item.errors.full_messages).to include("Price must be less than or equal to 9999999")
+      end
+
+      it 'priceが整数でない場合保存できない' do
```
商品価格についてのテストですが、
テストの正確性を向上させるために以下のテストを追加しましょう。
```
priceがからでは出品できない
priceに半角数字以外が含まれていると出品できない
```

OhyaYouki commented on spec/models/item_spec.rb:
```diff
@@ -0,0 +1,83 @@
+        @item.price = '1000円'
+        @item.valid?
+        expect(@item.errors.full_messages).to include("Price is not a number")
+      end
```
以下のテストを追加しましょう。
```
userが紐づいていないと出品できない
```

OhyaYouki reviewed (commented):
レビュー依頼ありがとうございます。

コメントいたしましたので、ご確認をお願いいたします。

コードの修正後、コミット・プッシュを行った上で再度レビュー依頼をお送りください。
ご対応のほど、よろしくお願いいたします。

shokoron committed:
修正しました

テスト画像
https://i.gyazo.com/e75f38197aef764137d281463d5d09ab.png

TakashiKozuma commented:
追加のレビュー依頼ありがとうございます！
修正点も含め、確認が完了しました。
LGTMです！！

こちらのブランチは忘れずにMergeいただき、ローカルのMasterにPullした後に次の実装のためのブランチを切ってから次にお進み下さい。
